packages/strapi-plugin-settings-manager/admin/src/components/WithFormSection/index.js

```diff
@@ -0,0 +1,124 @@
+/**
+*
+* WithFormSection
+*
+*/
+
+import React from 'react';
+import { forEach, has, isObject } from 'lodash';
+
+import InputNumber from 'components/InputNumber';
+import InputText from 'components/InputText';
+import InputToggle from 'components/InputToggle';
+import InputSelect from 'components/InputSelect';
+import InputEnum from 'components/InputEnum';
+import config from './config.json';
+import styles from './styles.scss';
+
+
+const WithFormSection = (InnerComponent) => class extends React.Component {
+  static propTypes = {
+    handleChange: React.PropTypes.func.isRequired,
+    section: React.PropTypes.oneOfType([
+      React.PropTypes.object,
+      React.PropTypes.array,
+    ]),
+    values: React.PropTypes.object,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      hasNestedInput: false,
+      showNestedForm: false,
+      inputWithNestedForm: '',
+    };
+  }
+
+  componentDidMount() {
+    // check if there is inside a section an input that requires nested input to display it on the entire line
+    if (isObject(this.props.section)) {
+      this.checkForNestedForm(this.props);
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.section !== this.props.section) {
+      this.setState({ showNestedForm: false, hasNestedInput: false, inputWithNestedForm: '' });
+      if (isObject(nextProps.section)) {
+        this.checkForNestedForm(nextProps);
+      }
+    }
+  }
+
+  checkForNestedForm(props) {
+    forEach(props.section.items, (input) => {
+      if(has(input, 'items')) {
+        this.setState({ hasNestedInput: true, inputWithNestedForm: input.target })
+
+        if (props.values[input.target]) {
+          this.setState({ showNestedForm: true });
+        }
+      }
+    });
+  }
+
+  handleChange = ({ target }) => {
+    // display nestedForm if the selected input has a nested form
+    if (target.name === this.state.inputWithNestedForm) {
+      this.setState({ showNestedForm: target.value });
+    }
+
+    this.props.handleChange({ target });
+  }
+
+  renderInput = (props, key) => {
+    const inputs = {
```

Comment:
This variable is static but because you put it here, everytime the `renderInput` method is called the variable is re-declared. Could it be interesting to use the `this` context? It's up to you, I'm just asking as usual :)